src/provider/model/ClientRegistry.ts

```diff
@@ -56,28 +56,31 @@ export class ClientRegistry {
     }
 
     /**
-     * Decides whether an app is running at the moment.
+     * Decides whether an app ready to receive `notification-action` events
      *
      * @param appUuid Uuid of the app.
      */
-    public isAppActive(appUuid: string): boolean {
-        return this._activeClients.some(client => client.uuid === appUuid);
+    public isAppActionReady(appUuid: string): boolean {
+        return this._actionReadyWindows.some(client => client.uuid === appUuid);
     }
 
     public getAllAppWindows(uuid: string): Identity[] {
-        return this._activeClients.filter((client) => {
-            return client.uuid === uuid;
-        });
+        return this._apiHandler.getClientConnections().filter((client: Identity) => client.uuid === uuid);
```

Comment:
To be clear, this function will return all windows with an established client connection. Is that definitely what it's meant to do as the function name seems to imply more than that (at least to me)? If we don't want to change the name, then may be a doc comment explaining this caveat.

Reply:
Thinking here was any windows that aren't connected don't matter anyway, but yeah, let's be explicit here

✅